plugin/hotswap-agent-spring-plugin/src/main/java/org/hotswap/agent/plugin/spring/reload/PropertiesChangedCommand.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2013-2023 the HotswapAgent authors.
+ *
+ * This file is part of HotswapAgent.
+ *
+ * HotswapAgent is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 2 of the License, or (at your
+ * option) any later version.
+ *
+ * HotswapAgent is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with HotswapAgent. If not, see http://www.gnu.org/licenses/.
+ */
+package org.hotswap.agent.plugin.spring.reload;
+
+import org.hotswap.agent.command.MergeableCommand;
+import org.hotswap.agent.command.Scheduler;
+import org.hotswap.agent.logging.AgentLogger;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.URL;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Add changed property to SpringChangedAgent.
+ */
+public class PropertiesChangedCommand extends MergeableCommand {
+    private static AgentLogger LOGGER = AgentLogger.getLogger(PropertiesChangedCommand.class);
+
+    ClassLoader appClassLoader;
+
+    URL url;
+    Scheduler scheduler;
+
+    public PropertiesChangedCommand(ClassLoader appClassLoader, URL url, Scheduler scheduler) {
```

Comment:
this looks not right to me, a scheduled command will schedule another command. I suggest simply call SpringChangedAgent's addChangedProperty() method in SpringPlugin#registerPropertiesListeners, and then schedule SpringChangedReloadCommand within the same method, which is 'addChangedProperty'.